paws_collector.js

```diff
@@ -13,19 +13,59 @@
 const async = require('async');
 const debug = require('debug')('index');
 const AWS = require('aws-sdk');
+const fs = require('fs');
 
 const AlAwsCollector = require('@alertlogic/al-aws-collector-js').AlAwsCollector;
 const m_packageJson = require('./package.json');
 
+const CREDS_FILE_PATH = '/tmp/paws_creds.json';
 var PAWS_DECRYPTED_CREDS = null;
 
-function getDecryptedPawsCredentials(callback) {
+function getPawsCredsFile(){
+    return new Promise((resolve, reject) => {
+        // check if the creds file is cached and retrieve it if it is not
+        if(!fs.existsSync(CREDS_FILE_PATH)){
+            const s3 = new AWS.S3({apiVersion: '2006-03-01'});
+            const kms = new AWS.KMS();
+
+            // doing the string manipulation here because doing it here is way less groos than doing it in the cfn
+            const s3Path = process.env.paws_s3_object_path.split('//')[1];
+            const s3PathParts = s3Path.split('/');
+
+            // retrive the object from S3
+            var params = {
+                Bucket: s3PathParts.shift(),
+                Key: s3PathParts.join('/')
+            };
+            s3.getObject(params, (err, data) => {
+                if (err) return reject(new Error(err, err.stack));
+
+                // encrypt the file contents and cache on the lambda container file system
+                const encryptParams ={
+                    Plaintext: data.Body,
+                    KeyId: process.env.paws_kms_key_arn
```

Comment:
is it ok to pass ARN here? Or is it required to use key id instead?

Reply:
My tests have successfully encrypted the file using an arn.